x-pack/plugins/ml/public/ml_url_generator/anomaly_detection_urls_generator.ts

```diff
@@ -193,6 +194,10 @@ export function createSingleMetricViewerUrl(
     mlTimeSeriesExplorer.functionDescription = functionDescription;
   }
 
+  if (zoom !== undefined) {
```

Comment:
Retesting this, I am finding sometimes the zoom isn't set correctly in the Single Metric Viewer. Could there be a race condition?

![Uploading view_series_time_range_bug2.gif…]()

Reply:
Thanks Pete. Seems like this is because the true `plotEarliest` and `plotLatest` can be updated  after the series are called. Fixed here  https://github.com/elastic/kibana/pull/86182/commits/c09e6f2734fbb23681550506937709c9a776d9a5.

Reply:
Tested latest edits in https://github.com/elastic/kibana/commit/c09e6f2734fbb23681550506937709c9a776d9a5 and the zoom is now set correctly for me every time :tada: